### PR TITLE
Add synthetic-data based benchmarking

### PR DESCRIPTION
**Pull request by juliusshufan**
To facilitate the performace evaludation, this PR add the synthetic-data based benchmarking, with specified number of iterations.
Changes are limited to verify_pretrained.py

### CODE REVIEW

**Comment by mli** [comment]
Job PR-767-1 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-767/1/index.html 
Code coverage of this PR:  vs. Master:

File: scripts/classification/imagenet/verify_pretrained.py [hunk 1]
```diff
@@ -38,6 +42,17 @@ def parse_args():
     opt = parser.parse_args()
     return opt
 
+
+def benchmark(network, ctx, batch_size=64, image_size=224, num_iter=100, datatype='float32'):
+    input_shape = (batch_size, 3) + (image_size, image_size)
+    data = mx.random.uniform(-1.0, 1.0, shape=input_shape, ctx=ctx, dtype=datatype)
+    tic = time.time()
+    for i in range(num_iter):
```
Comment (xinyu-intel) [inline scripts/classification/imagenet/verify_pretrained.py]: add warmup loops

File: scripts/classification/imagenet/verify_pretrained.py [hunk 2]
```diff
@@ -120,7 +143,7 @@ def test(ctx, val_data, mode='image'):
         _, top1 = acc_top1.get()
         _, top5 = acc_top5.get()
         return (1-top1, 1-top5)
-
+
```
Comment (xinyu-intel) [inline scripts/classification/imagenet/verify_pretrained.py]: remove this

File: scripts/classification/imagenet/verify_pretrained.py [hunk 3]
```diff
@@ -68,6 +83,14 @@ def parse_args():
         net.hybridize(static_alloc=True, static_shape=True)
     else:
         net.hybridize()
+    
+    if opt.benchmark:
+        print('-----benchmark mode for model %s-----'%opt.model)
+        time_cost = benchmark(network=net, ctx=ctx[0], image_size=opt.input_size, batch_size=opt.batch_size,
+            num_iter=opt.num_batches, datatype='float32')
+        fps = (opt.batch_size*opt.num_batches)/time_cost
+        print('With batch size %s, %s batches, inference performance is %.2f image/s' % (opt.batch_size, opt.num_batches, fps))
```
Comment (xinyu-intel) [inline scripts/classification/imagenet/verify_pretrained.py]: img/sec

**Comment by mli** [comment]
Job PR-767-2 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-767/2/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-767-3 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-767/3/index.html 
Code coverage of this PR:  vs. Master:

File: scripts/classification/imagenet/verify_pretrained.py (hunk 3 above)
Comment (juliusshufan) [reply scripts/classification/imagenet/verify_pretrained.py]: revised

**Comment by mli** [comment]
Job PR-767-4 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-767/4/index.html 
Code coverage of this PR:  vs. Master:

**Comment by xinyu-intel** [comment]
@juliusshufan Rebase and trigger the ci.

**Comment by mli** [comment]
Job PR-767-6 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-767/6/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-767-7 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-767/7/index.html 
Code coverage of this PR:  vs. Master:

File: scripts/classification/imagenet/verify_pretrained.py [hunk 4]
```diff
@@ -120,7 +144,7 @@ def test(ctx, val_data, mode='image'):
         _, top1 = acc_top1.get()
         _, top5 = acc_top5.get()
         return (1-top1, 1-top5)
-
+
```
Comment (xinyu-intel) [inline scripts/classification/imagenet/verify_pretrained.py]: lint

File: scripts/classification/imagenet/verify_pretrained.py [hunk 5]
```diff
@@ -38,6 +42,18 @@ def parse_args():
     opt = parser.parse_args()
     return opt
 
+def benchmark(network, ctx, batch_size=64, image_size=224, num_iter=100, datatype='float32'):
+    input_shape = (batch_size, 3) + (image_size, image_size)
+    data = mx.random.uniform(-1.0, 1.0, shape=input_shape, ctx=ctx, dtype=datatype)
+    dryrun = 5
+    for i in range(num_iter+dryrun):
```
Comment (xinyu-intel) [inline scripts/classification/imagenet/verify_pretrained.py]: lint

**Comment by mli** [comment]
Job PR-767-8 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-767/8/index.html 
Code coverage of this PR:  vs. Master:

**Pull request merged**